twitter-as.js

```diff
@@ -0,0 +1,232 @@
+var http = require("http");
+
+var MatrixTwitter = require("./mtwitter.js").MatrixTwitter;
+var https = require('https');
+var Cli = require("matrix-appservice-bridge").Cli;
+var Buffer = require("buffer").Buffer;
+var Bridge = require("matrix-appservice-bridge").Bridge;
+var RemoteUser = require("matrix-appservice-bridge").RemoteUser;
+var RemoteRoom = require("matrix-appservice-bridge").RemoteRoom;
+var RemoteRoom = require("matrix-appservice-bridge").MatrixRoom;
+var AppServiceRegistration = require("matrix-appservice-bridge").AppServiceRegistration;
+var Twitter;
+
+function roomPowers(users) {
+    return {
+        "content": {
+            "ban": 50,
+            "events": {
+                "m.room.name": 100,
+                "m.room.power_levels": 100,
+                "m.room.topic": 100,
+                "m.room.join_rules": 100,
+                "m.room.avatar": 100,
+                "m.room.aliases": 75,
+                "m.room.canonical_alias": 75
+            },
+            "events_default": 10,
+            "kick": 75,
+            "redact": 75,
+            "state_default": 0,
+            "users": users,
+            "users_default": 0
+        },
+        "state_key": "",
+        "type": "m.room.power_levels"
+    };
+}
+
+new Cli({
+    registrationPath: "twitter-registration.yaml",
+    bridgeConfig: {
+        schema: "config.yaml",
+        defaults: {
+            test: "ABC"
+        }
+    },
+    generateRegistration: function(reg, callback) {
+        reg.setId(AppServiceRegistration.generateToken());
+        reg.setHomeserverToken(AppServiceRegistration.generateToken());
+        reg.setAppServiceToken(AppServiceRegistration.generateToken());
+        reg.setSenderLocalpart("twitbot");
+        reg.addRegexPattern("users", "@twitter_.*", true);
+        reg.addRegexPattern("aliases", "#twitter_@.*", true);
+        callback(reg);
+    },
+    run: function(port, config) {
+        bridge = new Bridge({
+            homeserverUrl: "http://localhost:8008",
+            domain: "localhost",
+            registration: "twitter-registration.yaml",
+            controller: {
+                onUserQuery: function(queriedUser) {
+                    var twitter_user;
+                    var image_uri;
+                    return Twitter.get_user_by_id(queriedUser.localpart.substr(8)).then( (tuser) => {
```

Comment:
State what `queriedUser.localpart.substr(8)` is doing:
```javascript
// strip off "twitter_"
return Twitter.get_user_by_id(queriedUser.localpart.substr(8)).then( (tuser) => {
```

Alternatively, don't use magic numbers and let V8 optimise it away:

```javascript
return Twitter.get_user_by_id(queriedUser.localpart.substr("twitter_".length)).then( (tuser) => {
```